### PR TITLE
added support for queries made for columns with dot(.) in names

### PR DESCRIPTION
**Pull request by Vaibhavav**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Have you added new tests to prevent regressions?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you update the typescript typings accordingly (if applicable)?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

### Description of change

If Table has columns with dot(.) in names and doing a join with include, table name is not added to the column in select queries which raises potential issues in many scenarios. 
I am using Postgres with columns having dot(.) in name and two tables have that same column. When I am using include the query is giving error of ambiguous column as the table name is not prefixed in the query due to the dot(.).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -21,4 +21,4 @@ site
 .vscode/
 esdoc
 node_modules
-lib
+#lib
```
Comment (sushantdhiman) [inline .gitignore:24]: Why this change in `.gitignore`, you should only work on `src` folder

Comment (Vaibhavav) [reply .gitignore:24]: My bad, I was testing it. Updating the PR with the change

Comment (Vaibhavav) [reply .gitignore]: @sushantdhiman I have updated the PR

**Comment by sushantdhiman** [comment]
I don't think this can be safely included in sequelize, good solution will be to quote table names every where properly. I can accept such a PR

**Comment by Vaibhavav** [comment]
Hi, generated query by sequelize is  SELECT "challenges"."_id", "challenges"."community", "challenges"."is_active", "challenges"."is_public", "challenges"."is_scheduled", "challenges"."is_hidden", **"publish_status.label", "publish_status.value", "publish_status.timestamp",** "challenges"."publish_status_timeline", "publish_duration.start", "publish_duration.end", "status.label", "status.value", "status.timestamp", "challenges"."status_timeline", "challenges"."display_order", "challenges"."entity_type", "challenges"."challenge_type", "challenges"."challenge_type_group", "challenges"."timezone", "duration.start", "duration.end", "challenges"."members", "challenges"."total_score", "challenges"."reward_pot", "challenges"."results", "challenges"."creating_user", "challenges"."creating_admin_user", "challenges"."createdon", "challenges"."modifiedon" FROM "challenges" AS "challenges" LEFT OUTER JOIN "challengetypes" AS "fk_challenge_type" ON "challenges"."challenge_type" = "fk_challenge_type"."_id" WHERE "challenges"."is_public" = true ORDER BY "challenges"."createdon" DESC LIMIT 30;

Whereas expected query is -> SELECT "challenges"."_id", "challenges"."community", "challenges"."is_active", "challenges"."is_public", "challenges"."is_scheduled", "challenges"."is_hidden",  **"challenges"."publish_status.label",  "challenges"."publish_status.value",  "challenges"."publish_status.timestamp",** "challenges"."publish_status_timeline", "publish_duration.start", "publish_duration.end", "status.label", "status.value", "status.timestamp", "challenges"."status_timeline", "challenges"."display_order", "challenges"."entity_type", "challenges"."challenge_type", "challenges"."challenge_type_group", "challenges"."timezone", "duration.start", "duration.end", "challenges"."members", "challenges"."total_score", "challenges"."reward_pot", "challenges"."results", "challenges"."creating_user", "challenges"."creating_admin_user", "challenges"."createdon", "challenges"."modifiedon" FROM "challenges" AS "challenges" LEFT OUTER JOIN "challengetypes" AS "fk_challenge_type" ON "challenges"."challenge_type" = "fk_challenge_type"."_id" WHERE "challenges"."is_public" = true ORDER BY "challenges"."createdon" DESC LIMIT 30;

@sushantdhiman I did not get what you meant by quoting table name every where.

**Comment by sushantdhiman** [comment]
@Vaibhavav I think you should first present your issue as a SSCCE. Later on we can discussion if it should be fixed and if yes then how :)
